eng/pipelines/templates/steps/build-test.yml

```diff
@@ -44,57 +41,39 @@ steps:
       arguments: -vet -skipBuild -filter '${{ parameters.ServiceDirectory }}'
       workingDirectory: '${{ parameters.GoWorkspace }}'
 
-  - ${{ if eq(parameters.RunTests, 'true') }}:
-    - pwsh: |
-        go get github.com/jstemmer/go-junit-report
-        go get github.com/axw/gocov/gocov
-        go get github.com/AlekSi/gocov-xml
-        go get github.com/matm/gocov-html
-        go get -u github.com/wadey/gocovmerge
-      displayName: "Install Coverage and Junit Dependencies"
-      workingDirectory: '${{parameters.GoWorkspace}}'
-
-    - template: /eng/common/testproxy/test-proxy-docker.yml
+  - template: /eng/common/testproxy/test-proxy-docker.yml
 
-    - pwsh: |
```

Comment:
This change is going to clash with https://github.com/Azure/azure-sdk-for-go/pull/15560. Please have a look and we can figure out a good way to merge them. We might even want to remove the get_test_dirs.ps1 completely now and use the new `Get-AllPackageInfoFromRepo` function directly in your run_tests script.